packages/ui/src/components/select/select.tsx

```diff
@@ -0,0 +1,570 @@
+import * as React from "react"
+import {
+  UseComboboxProps,
+  UseComboboxReturnValue,
+  UseComboboxState,
+  UseComboboxStateChange,
+  UseComboboxStateChangeOptions,
+  useCombobox,
+} from "downshift"
+import { isEqual } from "lodash"
+import {
+  CheckMini,
+  ChevronUpDown,
+  EllipseMiniSolid,
+  MagnifyingGlassMini,
+  XMarkMini,
+} from "@medusajs/icons"
+import { DropdownMenu } from "@/components/dropdown-menu"
+import { clx } from "@/utils/clx"
+import { Badge } from "../badge"
+import { labelVariants } from "../label"
+
+const ALLOWED_SEARCH_KEYDOWN_CODES = ["Enter", "Escape", "ArrowUp", "ArrowDown"]
+const SCROLL_TOLERANCE = 30
+
+type SelectState<T> =
+  | ({
+      multi: boolean
+      selectedItems?: SelectItem[]
+      clearSelectedItems: () => void
+      selectAll: () => void
+      allSelected: boolean
+      search: boolean
+      onSearch: (searchTerm: string) => void
+      addItem: (item: SelectItem) => void
+      removeItem: (item: SelectItem) => void
+      onScrollToBottom: () => void
+    } & UseComboboxReturnValue<T>)
+  | null
+
+export type SelectItem = {
+  value: any
+  label: string
+  [k: string]: any
+}
+
+interface OnSelectChange {
+  (value: SelectItem): void
+  (value: SelectItem[]): void
+}
+
+export type SelectProps = {
+  multi?: boolean
+  search?: boolean
+  onChange: OnSelectChange
+  onSearch?: (searchTerm: string) => void
+  onScrollToBottom?: () => void
+} & React.ComponentPropsWithoutRef<typeof DropdownMenu>
+
+export const SelectContext = React.createContext<SelectState<SelectItem>>(null)
```

Comment:
```suggestion
export const SelectContext = React.createContext<SelectState<SelectItem> | null>(null)
```

And remove the `| null` from the SelectState definition